ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/name/vch-creation-wizard-name.component.spec.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ Copyright 2017 VMware, Inc. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+import {VchCreationWizardNameComponent} from './vch-creation-wizard-name.component';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {ReactiveFormsModule} from '@angular/forms';
+import {ClarityModule} from 'clarity-angular';
+import {HttpModule} from '@angular/http';
+import {CreateVchWizardService} from '../create-vch-wizard.service';
+import {Observable} from 'rxjs/Observable';
+
+describe('VchCreationWizardNameComponent', () => {
+
+  const ValidVCHName = 'unique-vm-container-host';
+  const InvalidVCHName = 'vm-container-host';
+
+  let component: VchCreationWizardNameComponent;
+  let fixture: ComponentFixture<VchCreationWizardNameComponent>;
+  let service: CreateVchWizardService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        ReactiveFormsModule,
+        HttpModule,
+        ClarityModule
+      ],
+      providers: [
+        {
+          provide: CreateVchWizardService,
+          useValue: {
+            checkVchNameUniqueness(name) {
+              if (name === 'vm-container-host') {
+                return Observable.of(false);
+              } else {
+                return Observable.of(true);
```

Comment:
So are we considering anything other than 'vm-container-host' a unique name? Why are we even checking for unique name I thought we were not checking that on front-end.

Reply:
This is a mock of the two possible values the service could respond with, so we can test the validity of the form, not the service. The service is in charge of checking uniqueness (it make a request to the API to accomplish this), but here we only have the form and its state for valid or invalid based on the service response. So with this mock we can test the final validity of the form (ant its fields). 'vm-container-hot' is only used as a *if block* branch selector here.